src/na/device/Generator.cpp

```diff
@@ -0,0 +1,380 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+/** @file
+ * @brief The MQT QDMI device generator for neutral atom devices.
+ */
+
+#include "na/device/Generator.hpp"
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <fstream>
+#include <istream>
+#include <nlohmann/detail/exceptions.hpp>
```

Comment:
As in the other file, the details headers should not be included (they also do not exist when using the single-header version of nlohman_json).